app/src/org/commcare/android/adapters/EntityDetailAdapter.java

```diff
@@ -24,18 +24,28 @@
  *
  */
 public class EntityDetailAdapter implements ListAdapter {
+
+    public interface EntityDetailViewModifier {
+        void modifyEntityDetailView(EntityDetailView edv);
+    }
     
-    Context context;
-    CommCareSession session;
-    Detail detail;
-    Entity entity;
-    DetailCalloutListener listener;
-    List<Integer> valid;
-    AudioController controller;
-    int detailIndex;
-    
-    public EntityDetailAdapter(Context context, CommCareSession session, Detail detail, Entity entity, 
-            DetailCalloutListener listener, AudioController controller, int detailIndex) {    
+    final Context context;
+    final CommCareSession session;
+    final Detail detail;
+    final Entity entity;
+    final DetailCalloutListener listener;
+    final List<Integer> valid;
+    final AudioController controller;
+    final int detailIndex;
```

Comment:
I'd like to hold off on doing big refactors like this in the middle of hte PR. They make it really hard to identify what was and was'nt a funcitonal change.